### DISCUSSION TITLE
What is the principle of converting an .mo file to a C language file? Is it done using a compiler for .mo files or is it interfaced with LLVM?

### DISCUSSION THREAD

**Discussion by pily1** [OpenModelica/OpenModelica#12411] [Q&A]
I want to konw the principle of converting a .mo file to a C language file
Is it done using a compiler for .mo files or is it interfaced with LLVM?
how can I debug the process when convert mo file to C language file?
what can I do can change the  generated C language content ?

**Comment by perost** [comment]
The compiler is `omc`, it supports compilation of both Modelica, MetaModelica, and Susan. To debug the compiler you can use the [MDT Eclipse plugin](https://openmodelica.org/doc/OpenModelicaUsersGuide/latest/mdt.html).

**Reply by pily1** [reply]
I have tried to install MDT Eclipse plugin,but the url is invalid now,"http://www.ida.liu.se/labs/pelab/modelica/OpenModelica/MDT“
when I enter the url to explorer,it shows "We are redirecting you to new our [www.OpenModelica.org ](https://www.openmodelica.org/index.php/home/tools/133)website!",then shows "Not Found
The requested URL was not found on this server.
Apache/2.4.52 (Ubuntu) Server at openmodelica.org Port 443"
![1111](https://github.com/OpenModelica/OpenModelica/assets/20216269/616dd2ed-d39e-414c-a0d0-d5e2bb831d18)
![2222](https://github.com/OpenModelica/OpenModelica/assets/20216269/276fc118-d67c-42ca-a4bb-6d91d7cab000)

**Reply by perost** [reply]
Did you try following the instructions and entering the URL in Eclipse? I don't think it's supposed to work in a web browser, only in Eclipse.

**Reply by pily1** [reply]
I have tried it in Eclipse,it has no responding,the following image is my operation process
![1111](https://github.com/OpenModelica/OpenModelica/assets/20216269/872a8e6f-7800-4dab-95fb-1cf8f95b47fe)
![2222](https://github.com/OpenModelica/OpenModelica/assets/20216269/7b421020-9fc8-49d7-b281-1a8345e95e0a)

**Reply by perost** [reply]
I think you need to uncheck the "Group items by category" check box for MDT to show up. It seems that step is missing in the instructions.

**Reply by pily1** [reply]
hi,I have tried it,maybe I have installed it successfully.
now I met a new problem,I click the "build project" button,it produceed nothing,the following is the result image.
how to configure the omc.exe?
![1111](https://github.com/OpenModelica/OpenModelica/assets/20216269/77562189-dc10-4df6-a537-ec0ee4f0850a)

**Reply by pily1** [reply]
I reset the OPENMODELICAHOME path,the following image is the path configure,after build,it produceed another problem.
![33333](https://github.com/OpenModelica/OpenModelica/assets/20216269/984ca6a0-6789-4127-a50e-87a0e555242b)
![44444](https://github.com/OpenModelica/OpenModelica/assets/20216269/2558a9f3-e965-47b0-916c-ff20fb633aa7)
this following image is the wrong dialog
![2222](https://github.com/OpenModelica/OpenModelica/assets/20216269/e8dedf18-6a21-429c-bc51-4a6b1b44eaae)
